torch/_inductor/lowering.py

```diff
@@ -1572,6 +1572,13 @@ def cat(inputs, dim=0):
             inputs, _ = require_channels_last(aten.cat, *inputs)
         return fallback_handler(aten.cat.default)(inputs, dim)
 
+    # When we have unbacked symints, we can't eagerly allocate the result
```

Comment:
mm if we don’t do this . is it possible that during the topological sort we just put whatever uses the unabacked in its right place ? or are we creating a cycle?
in other words maybe the real underlying issue is not detecting the dependency later on